pymatgen/io/vasp/inputs.py

```diff
@@ -663,14 +663,10 @@ def set_temperature(self, temperature: float):
         self.structure.add_site_property("velocities", velocities.tolist())
 
 
-with open(f"{module_dir}/incar_parameters.json") as json_file:
+with open(f"{module_dir}/incar_parameters.json", encoding="utf-8") as json_file:
     incar_params = json.loads(json_file.read())
 
 
-class BadIncarWarning(UserWarning):
-    """Warning class for bad Incar parameters."""
```

Comment:
I didn't see much point in defining a UserWarning when it's only called three times within the same method (`check_params`).